// streamango-scraper/dist/StreamangoScraper.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const StreamangoSource_1 = require("./StreamangoSource");
const source_scraper_core_1 = require("source-scraper-core");
const source_scraper_puppeteer_runner_1 = require("source-scraper-puppeteer-runner");
class StreamangoScraper extends source_scraper_core_1.SourceRunnerScraper {
    constructor() {
        super(...arguments);
        this.name = 'streamango';
        this.domains = ['streamango.com'];
        this.urlPattern = /(?:(?:https?:)?\/\/)?(?:[^.]+\.)?streamango\.com\/embed\/(\w+)/i;
        this.runner = new source_scraper_puppeteer_runner_1.PuppeteerRunner();
        this.defaultOptions = {};
    }
    async execWithArgs({ page }) {
        let srces;
        const srcs = await page.evaluate(() => {
            for (const script of document.getElementsByTagName('script')) {
                if (script.src)
                    continue;
                if (script.text.includes('srces.push')) {
                    eval(script.innerText);
                    return srces;
                }
            }
        });
        return ({
            sources: srcs.map((s) => new StreamangoSource_1.StreamangoSource({
                url: s.src,
                type: s.type,
                resolution: s.height,
                height: s.height,
                bitrate: s.bitrate
            }))
        });
    }
}
exports.StreamangoScraper = StreamangoScraper;
